### PR TITLE
kustomize 1.0.6 (new formula)

### PR DESCRIPTION
**Pull request by ahmetb**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by ahmetb** [comment]
> Formulae should not have a HEAD or devel spec

I don't understand this comment. Isn't HEAD allowed anymore? It exists on quite a few formulae.

**Comment by DomT4** [comment]
> Isn't HEAD allowed anymore?

Not for new formulae; It essentially lessens the support burden of new formulae. It's something that can be considered down the line if there's sustained interest in having it available.

File: Formula/kustomize.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      :revision => "017c4ae0aa19195db2a51ecc5aa82c56a1f1c99b"
+  head "https://github.com/kubernetes-sigs/kustomize.git"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    ENV["CGO_ENABLED"] = "0"
+
+    revision = `git rev-parse HEAD`.strip
```
Comment (DomT4) [inline Formula/kustomize.rb]: Use `Utils.popen_read()` rather than backticks please.

File: Formula/kustomize.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    tag = `git describe --tags`.strip
+    dir = buildpath/"src/github.com/kubernetes-sigs/kustomize"
+    dir.install buildpath.children - [buildpath/".brew_home"]
+    cd dir do
+      system "go", "install", "-ldflags",
+             "-s -X github.com/kubernetes-sigs/kustomize/pkg/commands.kustomizeVersion=#{tag}"\
```
Comment (DomT4) [inline Formula/kustomize.rb]: Use [this kind of style](https://github.com/Homebrew/homebrew-core/blob/de2d8e85c4ec432c3b9742bb3a970dc9ea0da13a/Formula/kubernetes-service-catalog-client.rb#L25-L29) please.

File: Formula/kustomize.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+             " -X github.com/kubernetes-sigs/kustomize/pkg/commands.gitCommit=#{revision}"
+      bin.install buildpath/"bin/kustomize"
+      prefix.install_metafiles
+    end
+  end
+
+  test do
+    assert_match "KustomizeVersion:", shell_output("#{bin}/kustomize version")
```
Comment (DomT4) [inline Formula/kustomize.rb]: Any chance of a test that exercises a deeper level of functionality? Thanks!

File: Formula/kustomize.rb [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  url "https://github.com/kubernetes-sigs/kustomize.git",
+      :tag => "v1.0.6",
+      :revision => "017c4ae0aa19195db2a51ecc5aa82c56a1f1c99b"
+  head "https://github.com/kubernetes-sigs/kustomize.git"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    ENV["CGO_ENABLED"] = "0"
```
Comment (DomT4) [inline Formula/kustomize.rb]: Isn't it off by default anyway? Maybe I'm misremembering.

File: Formula/kustomize.rb (hunk 1 above)
Comment (ahmetb) [reply Formula/kustomize.rb]: Done.

File: Formula/kustomize.rb (hunk 2 above)
Comment (ahmetb) [reply Formula/kustomize.rb]: Done.

File: Formula/kustomize.rb (hunk 3 above)
Comment (ahmetb) [reply Formula/kustomize.rb]: I thought about this, but unfortunately cannot find anything more meaningful.

File: Formula/kustomize.rb (hunk 4 above)
Comment (ahmetb) [reply Formula/kustomize.rb]: No. This is required to enable static compilation when the OS you're building on is the same as the target OS (GOOS). In this case both are `darwin`, so this is needed.

File: Formula/kustomize.rb (hunk 3 above)
Comment (DomT4) [reply Formula/kustomize.rb]: Any chance we could do something like an extremely minimal version of the first [usage example](https://github.com/kubernetes-sigs/kustomize#1-make-a-kustomization-file) (excluding the cluster part) or?

File: Formula/kustomize.rb (hunk 4 above)
Comment (DomT4) [reply Formula/kustomize.rb]: Super.

File: Formula/kustomize.rb (hunk 3 above)
Comment (ahmetb) [reply Formula/kustomize.rb]: Done, check it out.

Comment (DomT4) [reply Formula/kustomize.rb]: Nice! Great work!

**Comment by DomT4** [comment]
Merged in https://github.com/Homebrew/homebrew-core/commit/ed932d411a3470be81d5009b0b654e7658503340. Thank you for your contribution to Homebrew @ahmetb; we appreciate it! :smiley_cat:

**Pull request closed without merging**